src/main/java/com/nutomic/syncthingandroid/util/Util.java

```diff
@@ -55,4 +65,58 @@ public static String readableTransferRate(Context context, long bits) {
         return new DecimalFormat("#,##0.#")
                 .format(bytes / Math.pow(1024, digitGroups)) + " " + units[digitGroups];
     }
+
+    /**
+     * Normally an application's data directory is only accessible by the corresponding application.
+     * Therefore, every file and directory is owned by an application's user and group. When running Syncthing as root,
+     * it writes to the application's data directory. This leaves files and directories behind which are owned by root having 0600.
+     * Moreover, those acitons performed as root changes a file's type in terms of SELinux.
+     * A subsequent start of Syncthing will fail due to insufficient permissions.
+     * Hence, this method fixes the owner, group and the files' type of the data directory.
+     * 
+     * @return true if the operation was successfully performed. False otherwise.
+     */
+    public static boolean fixAppDataPermissions(Context context) {
+        // We can safely assume that root magic is somehow available, because readConfig and saveChanges check for
+        // read and write access before calling us.
+        // Be paranoid :) and check if root is available.
+        // Ignore the 'use_root' preference, because TODO
+        if (!Shell.SU.available()) {
+            Log.e(TAG,"Root is not available. Cannot fix permssions ");
+            return false;
+        }
+
+        try {
+            ApplicationInfo appInfo = context.getPackageManager().getApplicationInfo(context.getPackageName(),0);
+            Log.d(TAG,"Uid of '" + context.getPackageName() + "' is " + appInfo.uid);
+            Process fixPerm = Runtime.getRuntime().exec("su");
+            DataOutputStream fixPermOut = new DataOutputStream(fixPerm.getOutputStream());
+            String dir = context.getFilesDir().getAbsolutePath();
+            String cmd = "chown -R " + appInfo.uid + ":" + appInfo.uid + " " + dir + "\n";
+            Log.d(TAG,"Running: '" + cmd);
+            fixPermOut.writeBytes(cmd);
+            // For now, we assume that a file's type should *always* be 'u:object_r:app_data_file:s0:c512,c768'.
+            // At least for those files residing in an application's data folder.
+            // Running syncthing as root, makes it 'u:object_r:app_data_file:s0'.
+            // Simply reverting the type to its default should do the trick.
```

Comment:
Is that comment still relevant, seeing as chown uses the uid directly?

Reply:
Yes, it is still relevant -- more or less. I'm rephrasing it.